### PR TITLE
Extended scan

### LINKED ISSUE #1008: Support for MultiRowRangeFilter
Taking a cursory glance at the `ScanAdapter`, it looks like the underlying `RowSet` supports multiple `RowRange`s. Would this make it possible to support HBase's [`MultiRowRangeFilter`](https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/filter/MultiRowRangeFilter.html)?

### PR DESCRIPTION
**Pull request by sduskis**
Allow users to set arbitrary keys and key ranges in a scan.  This will partially fix #1008.

### CODE REVIEW

File: bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java [hunk 1]
```diff
@@ -78,4 +103,31 @@ public void maxVersionsIsSet() {
             .build(),
         rowRequestBuilder.getFilter().getChain());
   }
+
+  @Test
+  public void testExtendedScan(){
+    byte[] row1 = Bytes.toBytes("row1");
+    byte[] row2 = Bytes.toBytes("row2");
+
```
Comment (igorbernstein2) [inline bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java]: Would be good to test the boundary cases for binary keys: trailing 0xff and negatives

File: bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java [hunk 2]
```diff
@@ -51,17 +54,39 @@ public ReadRowsRequest applyPreSendHook(ReadRowsRequest readRowsRequest) {
...
+    byte[] prefixEnd = calculatePrefixEnd(prefix);
     Scan scan = new Scan();
-    scan.setStartRow(startKey);
-    scan.setStopRow(stopKey);
+    scan.setRowPrefixFilter(prefix);
     ReadRowsRequest.Builder request = scanAdapter.adapt(scan, throwingReadHooks);
-    RowRange range = request.getRows().getRowRanges(0);
-    Assert.assertArrayEquals(startKey, range.getStartKeyClosed().toByteArray());
-    Assert.assertArrayEquals(stopKey, range.getEndKeyOpen().toByteArray());
+    Assert.assertEquals(toRowSet(toRange(prefix, prefixEnd)), request.getRows());
```
Comment (igorbernstein2) [inline bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java]: Would be good to test the boundary cases for binary keys: trailing 0xff and negatives

**Comment by igorbernstein2** [comment]
LGTM

File: bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java (hunk 2 above)
Comment (sduskis) [inline bigtable-hbase-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/adapters/read/TestScanAdapter.java]: good call.  Done.

**Comment by igorbernstein2** [comment]
LGTM
